third_party/opencv_ios_source.bzl

```diff
@@ -0,0 +1,150 @@
+                        platform, _OPENCV_FRAMEWORK_DIR_NAME ,binary_name))
+                binaries_to_symlink.append((symlink_output, file))
+
+    return output_files, binaries_to_symlink
+
+def _unzip_opencv_xcframework_impl(ctx):
+
+    # Array to iterate over the various platforms to declare output files and
+    # symlinks.
+    platform_filepath_lists = [
+        (_OPENCV_SIMULATOR_PLATFORM_DIR_NAME, OPENCV_XCFRAMEWORK_IOS_SIMULATOR_FILE_PATHS),
+        (_OPENCV_DEVICE_PLATFORM_DIR_NAME, OPENCV_XCFRAMEWORK_IOS_DEVICE_FILE_PATHS)
+    ]
+    
+    # Gets an exhaustive list of output files which are  present in the xcframework.
+    # Also gets array of `(binary simlink, binary)` pairs which are to be symlinked
+    # using `ctx.actions.symlink()`.
+    output_files, binaries_to_symlink = _opencv2_file_list(ctx, platform_filepath_lists)
+    output_files.append(ctx.actions.declare_file(OPENCV_XCFRAMEWORK_INFO_PLIST_PATH))
+
+    # xcframeworks have a directory structure in which the `opencv2.framework` folders for each 
+    # platform contain directories which are symlinked to the respective folders of the version
+    # in use. Simply unzipping the zip of the framework will not make Bazel treat these
+    # as symlinks. They have to be explicity declared as symlinks using `ctx.actions.declare_symlink()`. 
+    directory_symlinks = _opencv2_directory_symlinks(
+        ctx, [_OPENCV_SIMULATOR_PLATFORM_DIR_NAME, _OPENCV_DEVICE_PLATFORM_DIR_NAME])
+    
+    output_files = output_files + directory_symlinks
+    
+    args = ctx.actions.args()
+
+    # Add the path of the zip file to be unzipped as an argument to be passed to
+    # `run_shell` action.
+    args.add(ctx.file.zip_file.path)
+
+    # Add the path to the directory in which the framework is to be unzipped to.
+    args.add(ctx.file.zip_file.dirname)
+
+    ctx.actions.run_shell(
+        inputs = [ctx.file.zip_file],
+        outputs = output_files,
+        arguments = [args],
+        progress_message = "Unzipping %s" % ctx.file.zip_file.short_path,
+        command = "unzip -qq $1 -d $2",
+    )
+    
+    # The symlinks of the opencv2 binaries for each platform in the xcframework 
+    # have to be symlinked using the `ctx.actions.symlink` unlike the directory
+    # symlinks which can be expected to be valid when unzipping is completed.
+    # Otherwise, when tests are run, the linker complaints that the binary is
+    # not found.
+    binary_symlink_files = []
+    for (symlink_output, binary_file) in binaries_to_symlink:
+        ctx.actions.symlink(output = symlink_output, target_file = binary_file)
+        binary_symlink_files.append(symlink_output)
+    
+    # Return all the declared output files and symlinks as the output of this
+    # rule.
+    return [ DefaultInfo(files=depset(output_files + binary_symlink_files)) ]
+
+
+# This rule unzips an `opencv2.xcframework.zip` created by a genrule that 
```

Comment:
@schmidt-sebastian We can write custom rules which output directories. But `apple_static_xcframework_import` cannot consume directories. It can only take a list of files. Hence we must declare a full list of all files expected to be in opencv2.xcframework in this custom rule which will provide the headers to `apple_static_xcframework_import`. So hardcoding them cannot be avoided. Since we are fixing opencv at 4.5.1, this won't cause any problems.